cocos/animation/core/pose.ts

```diff
@@ -104,7 +104,9 @@ export function blendTransformsInto (
             blendIntoTransformArrayAt(target, source, alpha, iTransform);
         }
     } else {
-        for (const involvedTransformIndex of transformFilter.involvedTransforms) {
+        // NOTE: cannot use for-of statement for Readonly ArrayBuffer on TS 4.2 for OH platform, wait for they upgrade ts version.
+        for (let index = 0; index < transformFilter.involvedTransforms.length; ++index) {
```

Comment:
Please add an issue for it too.

Reply:
added https://github.com/cocos/cocos-engine/issues/14715